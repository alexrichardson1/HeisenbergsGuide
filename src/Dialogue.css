.table {
  background-color: white;
  border: yellowgreen;
  border-width: 10px;
  display: table;
  margin-left: 50px;
}

input {
  border: 2px solid;
  border-radius: 4px;
  font-size: 1rem;
  margin: 0.25rem;
  min-width: 125px;
  padding: 0.5rem;
  display: block;
  margin-left: 50px;
  /* clear: left; */
}
