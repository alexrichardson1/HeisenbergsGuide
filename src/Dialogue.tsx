import { useEffect, useState } from "react";
import { dialogues } from "./dialogues";

var id = 0;

function Dialogue() {
  document.title = "Bain's Dialogues";
  const [search, setSearch] = useState("");
  const [filteredDialogues, setFilteredDialogues] = useState(() => dialogues);

  useEffect(() => {
    setFilteredDialogues(
      dialogues.filter((d) => d.toLowerCase().includes(search.toLowerCase()))
    );
  }, [search]);

  return (
    <div>
      <h1>Dialogue Lists</h1>
      <input
        type="text"
        placeholder="Enter Bain's Dialogue Here"
        onChange={(e) => setSearch(e.target.value)}
      />

      {filteredDialogues.map((d) => (
        <div key={id++}>{d}</div>
      ))}
    </div>
  );
}

export default Dialogue;
