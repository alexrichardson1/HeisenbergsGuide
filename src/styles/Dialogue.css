.table {
  display: table;
  color: black;
  font-family: "Raleway", sans-serif;
  font-weight: 800;
  line-height: 30px;
  background-color: white;
  border: 2px solid blue;
  border-radius: 30px;
  border-width: thick;
  padding: 5px;
}

input {
  border: 2px solid blue;
  border-radius: 4px;
  border-width: thick;
  margin: 0.25rem;
  padding: 0.5rem;
  display: block;
  margin-bottom: 10px;
  outline: none;
}

@media (min-width: 481px) {
  .table {
    margin-left: 50px;
    font-size: 20px;
  }

  input {
    font-size: 15pt;
    min-width: 125px;
    margin-left: 50px;
  }
}

@media (max-width: 480px) {
  .table {
    margin-left: atuo;
    font-size: medium;
  }

  input {
    font-size: 20pt;
    min-width: 350px;
  }
}
