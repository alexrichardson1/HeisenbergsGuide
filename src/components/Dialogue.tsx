import { useEffect, useState } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { dialogues } from "../dialogues";
import "styles/Dialogue.css";

function Dialogue(): JSX.Element {
  let id = 0;
  const [search, setSearch] = useState("");
  const [filteredDialogues, setFilteredDialogues] = useState(() => dialogues);

  useEffect(() => {
    document.title = "Bain's Dialogues";
  }, []);

  useEffect(() => {
    setFilteredDialogues(
      dialogues.filter((d) => d.toLowerCase().includes(search.toLowerCase()))
    );
  }, [search]);

  return (
    <div>
      <h1>Dialogue Lists</h1>
      <input
        type="text"
        placeholder="Enter Bain's Dialogue Here"
        onChange={(e) => setSearch(e.target.value)}
      />
      <span className="table">
        {filteredDialogues.map((d) => (
          <div key={id++}>{d}</div>
        ))}
      </span>
      <LinkContainer to="/">
        <section className="container">
          <button data-hover="Go Back">
            <div>Cook Off Guide</div>
          </button>
        </section>
      </LinkContainer>
    </div>
  );
}

export default Dialogue;
