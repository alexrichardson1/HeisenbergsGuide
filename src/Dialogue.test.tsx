import renderer from "react-test-renderer";
import Dialogue from "./components/Dialogue";

test("renders correctly", () => {
  const tree = renderer.create(<Dialogue />).toJSON();
  expect(tree).toMatchSnapshot();
});
