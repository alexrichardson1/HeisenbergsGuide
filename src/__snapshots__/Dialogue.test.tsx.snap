// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Dialogue Lists
  </h1>
  <input
    onChange={[Function]}
    placeholder="Enter Bain's Dialogue Here"
    type="text"
  />
  <span
    className="table"
  >
    <div>
      Now we need... I'm going with (ingredient). Hold up! ❌
    </div>
    <div>
      Uh... mhm...We'll go with (ingredient). Nope, that's not it! ❌
    </div>
    <div>
      Alright, next ingredient...(ingredient)? Sorry, that wasn't right. ❌
    </div>
    <div>
      Alright, let's see...We'll go with (ingredient). Strike that. ❌
    </div>
    <div>
      Hold on... I'm going with (ingredient). Wait a minute! ❌
    </div>
    <div>
      Uh, I hope I got this right... (ingredient)? ...I'm XX% sure. ✅
    </div>
    <div>
      Uh. I hope I got this right... (ingredient)? You got any of that lying around? ✅
    </div>
    <div>
      Uh. I hope I got this right... We'll go with (ingredient). ...I...think. ✅
    </div>
    <div>
      Okay, what's next... (ingredient)? ...I'm XX% sure. ✅
    </div>
    <div>
      Okay, what's next... Oh I'm going with (ingredient). ...it's a toss-up, really. ✅
    </div>
    <div>
      Okay, what comes after that... We'll go with (ingredient). That should get the process going. ✅
    </div>
    <div>
      Okay, what comes after that... (ingredient). That should get the process going. ✅
    </div>
    <div>
      Okay, what comes after that... It's (ingredient). That should get it on. ✅
    </div>
    <div>
      Uh... mhm... It's (ingredient). ...that's my very best guess. ✅
    </div>
    <div>
      Uh... mhm... (ingredient). ...this website better be accurate. ✅
    </div>
    <div>
      Uh... mhm... (ingredient). That should get it on. ✅
    </div>
    <div>
      Uh... mhm... (ingredient)? ...IF my calculations are correct. ✅
    </div>
    <div>
      Uh... mhm... (ingredient)? ...and kiss your lucky charms. ✅
    </div>
    <div>
      Uh... mhm... I'm going with (ingredient). Go for it. ✅
    </div>
    <div>
      Alright, next ingredient... We'll go with (ingredient). That should get the process going. ✅
    </div>
    <div>
      Alright, next ingredient... We'll go with (ingredient). ...I'm XX% sure. ✅
    </div>
    <div>
      Alright, let's see... (ingredient). ...I think so, at least. ✅
    </div>
    <div>
      Alright, let's see... (ingredient). ...I think so, at least. ✅
    </div>
    <div>
      Alright, let's see... (ingredient)? ...yeah... yeah. ✅
    </div>
    <div>
      Alright, let's see... (ingredient)? ...I'm XX% sure. ✅
    </div>
    <div>
      Alright, let's see... It's (ingredient). ...unless I've missed something... ✅
    </div>
    <div>
      Uh, I hope I got this right... (ingredient)? You got any of that lying around? ✅
    </div>
    <div>
      Uh, I hope I got this right... We'll go with (ingredient). ...for sure. Or, most likely. ✅
    </div>
    <div>
      Now we need... It's (ingredient). ...that's my best guess. ✅
    </div>
    <div>
      Now we need... We'll go with (ingredient). ...I'm XX% sure. ✅
    </div>
    <div>
      Now we need... We'll go with (ingredient). ...what's the worst that could happen, right? ✅
    </div>
    <div>
      Now we need... I'm going with (ingredient). ...these internet descriptions are iffy at best. ✅
    </div>
    <div>
      Hold on... (ingredient). I'm XX% sure. ✅
    </div>
    <div>
      Uh, if I read this correctly... uh... Oh I'm going with (ingredient). ...I'm XX% sure. ✅
    </div>
    <div>
      It's supposed to be (ingredient)! Pour it in. ❌✅
    </div>
    <div>
      It's supposed to be (ingredient)! That should get the process going. ❌✅
    </div>
    <div>
      I messed up. It should'a been (ingredient). That should get it on. ❌✅
    </div>
    <div>
      I messed up. It should'a been (ingredient). Pour it in. ❌✅
    </div>
    <div>
      We need (ingredient)! ❌✅
    </div>
    <div>
      That should'a been (ingredient)! ...unless I've missed something... ❌✅
    </div>
    <div>
      ...(ingredient)?...I'm XX% sure. ❌✅
    </div>
    <div>
      By method of elimination it has to be (ingredient). ...what's the worst that could happen, right? ❌✅
    </div>
    <div>
      My bad. It's (ingredient). Add some. ❌✅
    </div>
  </span>
</div>
`;
